### DISCUSSION TITLE
Third person with Controller camera control (using the right joystick)?

### DISCUSSION THREAD

**Discussion by Patataonk** [ramokz/phantom-camera#242] [General]
Hi
has anyone gotten phantom camera to work with a right stick controller to control the camera?
I just can't seem to get it to behave well as the mouse camera control

heres my code attempt:

```
		var pcam_rotation_degrees: Vector3
		var input_dir = Input.get_vector("cam_rotation_left", "cam_rotation_right", "cam_up", "cam_down")

		pcam_rotation_degrees.y += input_dir.x *1.2
		pcam_rotation_degrees.y = wrapf(pcam_rotation_degrees.y, min_pitch, max_pitch)
		pcam_rotation_degrees.y = 0
		pcam_rotation_degrees.x += input_dir.y *0.2
		pcam_rotation_degrees.x = clampf(pcam_rotation_degrees.x, min_yaw+5, max_yaw +50)
		
		pcam.set_third_person_rotation_degrees(pcam_rotation_degrees + _player_pcam.get_third_person_rotation_degrees())
```

**Comment by Patataonk** [comment]
had a deeper look and it seems to be a godot bug

https://github.com/godotengine/godot/issues/81816

**Reply by ramokz** [reply]
Interesting, is it the same controller model and connection setup that you're using?

**Reply by Patataonk** [reply]
I have the original XBOX One bluetooth controller

When I connect wirelessly via bluetooth I get the joystick bug, but it works fine when wired

**Reply by ramokz** [reply]
That sounds odd. By the looks of it, it should be merged into 4.2, but you're still encountering the issue?

**Reply by Patataonk** [reply]
yes
im using 4.3 dev5

**Reply by ramokz** [reply]
Is it a case of where it doesn't register any input at all, or where the input isn't aligning (value or direction-wise) with what you want?

**Reply by Patataonk** [reply]
input doesnt align

So if I move right joystick up it registers and joystick left
down is right
left is up
right is down


I also have the same behavious as this https://github.com/godotengine/godot/issues/87112#issuecomment-1980645624
where if the window is unfocused the right joystick works normally wirelessly

**Reply by ramokz** [reply]
Strange, but yes, this seems to be a Godot rather than an addon issue.
